### PR TITLE
chore: upgrade `@types/react` to 18.2.37 to have `plaintext-only`

### PR DESCRIPTION
**Pull request by cromoteca**
This seems to help building a version of React Components that is up to date with changes in `@types/react`, especially including `plaintext-only` as an option for `contentEditable`.

The problem seems to come from https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67250. Our React wrappers are generated using an older version of `@types/react`, so there's a type conflict if the latest version of that library is used in a Hilla project.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -534,6 +534,8 @@
     "./RichTextEditor": "./RichTextEditor.js",
     "./Scroller": "./Scroller.js",
     "./Select": "./Select.js",
+    "./SideNav": "./SideNav.js",
+    "./SideNavItem": "./SideNavItem.js",
```
Comment (tomivirkki) [inline package.json:538]: nit: this seems like an unrelated change

**Review by tomivirkki — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (cromoteca) [reply package.json:538]: That happened automatically.
